### PR TITLE
Port Dominations to 2.9.1

### PR DESCRIPTION
**Pull request by PrOF-kk**
Compatible only as client with 2.9.1 servers, it won't sync existing dominations
The scoreboard medals suffer from some scaling issues
Adding a Revenge medal would be very useful. I have the code ready, but creating a 9x11 sprite for it will be kind of a challenge
After this is merged, (hopefully), we'll really have to refactor ScoreboardController's draw event

Original code from Vindicator, port and fixes by the commit author(s)

### CODE REVIEW

**Comment by Kostia4381** [comment]
Are you **sure** you're pullrequesting to the right repo? hehe

**Comment by PrOF-kk** [comment]
This should be ready for merging. Even though I didn't get any feedback from 2.9.2 Beta 2, I only added a visual indicator for nemesis, so it should work fine if it did before. The ScoreboardController Draw event is still a mess (aka not truly aspect ratio independent), but it should be easy to refactor in another PR, starting after merging this to avoid merge conflicts

File: Source/gg2/Scripts/Killtable/killtable_clear.gml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+//Clears the given killtable
+argument0 = killtable_create();
```
Comment (Medo42) [inline Source/gg2/Scripts/Killtable/killtable_clear.gml]: You can't replace the contents of a variable for the caller, because that would be call-by-reference and GM does not have that. All you are doing is setting the local variable called argument0 to a new killtable, and that variable will stop existing after the script returns, so nobody will ever see (or destroy) the new killtable again, and the caller will not see any effect at all.

Don't bother fixing it though, I'm replacing the killtable with a ds_map. You should look at that structure, most programming languages offer something like it.

File: Source/gg2/Scripts/Killtable/killtable_delete.gml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+//Removes a player and its corresponding value
+//Arg0: Killtable
+//Arg1: Player
+player_list = ds_list_find_value(argument0, 0);
+value_list = ds_list_find_value(argument0, 1);
+index = ds_list_find_index(player_list, argument1);
```
Comment (Medo42) [inline Source/gg2/Scripts/Killtable/killtable_delete.gml]: Just as feedback (again, I'll fix it myself): You should use local variables (var keyword) for anything that is only needed within one script, otherwise you store it in whatever instance is currently selected, possibly overwriting variables that are already there (especially nasty in one place where you have a variable called "player" since we use that name all over the place for instance variables). This is something I noticed all over the code.

File: Source/gg2/Scripts/Sound/calculateVolume.gml [hunk 3]
```diff
@@ -3,13 +3,22 @@
...
\ No newline at end of file
+    */
+    global.dist = dist;
+    if (dist > 10) {
+        return (1500/dist);
+    }
+    else {
+        return 1;
+    }
+}
```
Comment (Medo42) [inline Source/gg2/Scripts/Sound/calculateVolume.gml]: Why is this changed?

Comment (PrOF-kk) [reply Source/gg2/Scripts/Sound/calculateVolume.gml]: Oh I remember now, this was an attempt at 1. Figuring put how the panning code worked, 2. Fix audio being based on the center of the screen, which causes issues when aiming as sniper
This should be removed, my bad

**Comment by PrOF-kk** [comment]
I've looked at MedalS, it shouldn't be where it is now, we should make a new folder for domination sprites

File: Source/gg2/Scripts/Killtable/killtable_clear.gml (hunk 1 above)
Comment (PrOF-kk) [reply Source/gg2/Scripts/Killtable/killtable_clear.gml]: Ok wow looks like the previous code used 3 ds_lists instead of a ds_map, much cleaner, I agree
